integration_test/secret_storage_test.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:concordium_wallet/entities/secret_box.dart';
+import 'package:concordium_wallet/exceptions/secret_storage_exception.dart';
+import 'package:concordium_wallet/providers/secret_storage.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:integration_test/integration_test.dart';
+
+/// Integration test class for secrect storage usages.
+///
+/// Test are implemented as integration test since implementation differs on web and
+/// mobile and integration test are run on all supported platforms in our CI.
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late SecretStorageProvider storage;
+
+  setUpAll(() async {
+    await Hive.initFlutter();
+    storage = await SecretStorageProviderFactory.create();
+  });
+
+  tearDown(() async {
+    if (kIsWeb) {
+      if (Hive.isBoxOpen(SecretBoxEntity.table)) {
+        await Hive.lazyBox<SecretBoxEntity>(SecretBoxEntity.table).clear();
+      }
+      if (Hive.isBoxOpen(WebSecretStorageProvider.encryptedBoxTable)) {
+        await Hive.lazyBox(WebSecretStorageProvider.encryptedBoxTable).clear();
+      }
+    } else {
+      const mobileStorage = FlutterSecureStorage(aOptions: AndroidOptions(encryptedSharedPreferences: true));
+      await mobileStorage.deleteAll();
+    }
+  });
+
+  tearDownAll(() async {
+    if (kIsWeb) {
+      try {
+        // [Hive.deleteFromDisk] doesn't close by itself on
+        // web https://github.com/isar/hive/pull/734.
+        await Hive.deleteFromDisk().timeout(const Duration(seconds: 2));
+      } catch (_) {}
+    }
+  });
+
+  // Runs before other test since box should
+  if (kIsWeb) {
+    testWidgets("Given no password, when read, then throw exception", (widgetTester) async {
+      // Arrange
+      const String key = "foo";
+
+      // Act
+      try {
+        await storage.read(key);
+        fail("Read should have triggered exception");
+      } on SecretStorageException catch (e) {
+        // Assert
+        expect(e.error, SecretStorageError.encryptedBoxNotOpened);
```

Comment:
Not sure I understand. It fails if none exception is thrown, and only `SecretStorageException` are caught.

Reply:
You are of course right - don't know how I missed the fail() on the next line :D

But maybe it still makes sense to capture the result in an expectedError variable, so that the test only has a single exit point / a single place where we do the asserts.